### ISSUE TITLE
IW debugger always breaks in site-packages/IPython/core/interactiveshell.py

### ISSUE THREAD

**Issue by janosh** [microsoft/vscode-jupyter#9815]
Clicking the code cell debug button always breaks on this line in `~/.venv/py310/lib/python3.10/site-packages/IPython/core/interactiveshell.py`:

```py
finally:
    # Reset our crash handler in place
    sys.excepthook = old_excepthook
```

![Screen Shot 2022-04-28 at 09 47 39](https://user-images.githubusercontent.com/30958850/165714936-1162ae38-5d3f-4366-9523-278208558b44.png)

![Screen Shot 2022-04-28 at 09 52 15](https://user-images.githubusercontent.com/30958850/165715561-5046b903-73b4-4711-acb2-aec73ee0ca63.png)

(regardless of "Uncaught Exceptions" being checked or unchecked). This has been previously mentioned in https://github.com/microsoft/vscode-jupyter/issues/6560#issuecomment-875186707 and with 10 upvotes on SO suggesting the problem is not unique to me https://stackoverflow.com/q/46495269.

**Comment by janosh** [comment]
Oh just noticed the SO post is about Jupyter in general and not specific to VSCode. So perhaps wrong place for this issue. Unless someone has advice on what debug config might fix this issue?

**Comment by IanMatthewHuff** [comment]
@janosh Are you possibly on an old ipykernel version? That's what was showing this issue previously.

**Comment by janosh** [comment]
@IanMatthewHuff I don't think so. I just ran `pip install -U ipykernel`

```sh
      Successfully uninstalled ipykernel-6.6.0
Successfully installed ipykernel-6.13.0 psutil-5.9.0
```

Tried debug cell again but same behavior.

**Comment by IanMatthewHuff** [comment]
Hmm, any chance you have turned off our Jupyter -> Debug Just My Code setting? I think that it usually breaks there if that setting is unchecked.

**Comment by janosh** [comment]
Yes, that's it! I don't use it that often but sometimes it's nice to be able to look e.g. at the shape and values of tensors in Pytorch. I guess I can just add `"jupyter.debugJustMyCode": false` for those debug sessions and drop it otherwise.

**Comment by IanMatthewHuff** [comment]
Sound good. It think that I'll close this issue for now. Though I'm wondering a tiny bit if this is a possible situation where we might need notification of some type? It's legitimately not obvious when you are in this situation and easy to forget. I did the same thing on a machine that I was testing with this week. Didn't remember that I had flipped the setting and was confused by the break location.
